clang/lib/Sema/SemaDecl.cpp

```diff
@@ -2862,6 +2862,10 @@ static bool mergeDeclAttribute(Sema &S, NamedDecl *D,
   else if (const auto *NT = dyn_cast<HLSLNumThreadsAttr>(Attr))
     NewAttr = S.HLSL().mergeNumThreadsAttr(D, *NT, NT->getX(), NT->getY(),
                                            NT->getZ());
+  else if (const auto *NT = dyn_cast<HLSLWaveSizeAttr>(Attr))
```

Comment:
This code block looks ripe for a refactoring to use a switch to detect the attribute type... We should consider doing that cleanup in a follow-up change.